stdlib/REPL/src/docview.jl

```diff
@@ -299,8 +301,33 @@ function summarize(io::IO, T::DataType, binding::Binding)
     end
 end
 
-function summarize(io::IO, m::Module, binding::Binding)
-    println(io, "No docstring found for module `", m, "`.\n")
+function find_readme(m::Module)::Union{String, Nothing}
+    isnothing(pathof(m)) && return nothing
+    path = dirname(pathof(m))
+    top_path = pkgdir(m)
+    while true
+        for readme in ["README.md", "readme.md", "Readme.md", "ReadMe.md"]
+            readme_path = joinpath(path, readme)
+            if isfile(readme_path)
+                return readme_path
+            end
+        end
+        path == top_path && break # go no further than pkgdir
+        path = dirname(path) # work up through nested modules
+    end
+    return nothing
+end
+function summarize(io::IO, m::Module, binding::Binding; nlines::Int = 200)
+    readme_path = find_readme(m)
+    if !isnothing(readme_path)
+        println(io, "No docstring found for module `", m, "`.\n")
+        readme_lines = readlines(readme_path)
```

Comment:
Why are we defining this up here, and now where it is used?
I am guessing because if it is empy we don't want to display anything?
```suggestion
readme_lines = readlines(readme_path)
isempty(readme_lines) && return  # don't say we are going to print empty file
```
May want to print something to that effect.

I would suggest using `eachline` instead of `readlines` since no need to read whole thing if only printing part.
but `eachline` is stateful, and checking if it `isempty` would consume the first line.